presto-spi/src/main/java/com/facebook/presto/spi/NodeState.java

```diff
@@ -15,7 +15,30 @@
 
 public enum NodeState
 {
-    ACTIVE,
-    INACTIVE,
-    SHUTTING_DOWN
+    ACTIVE(1),
+    INACTIVE(2),
+    SHUTTING_DOWN(3);
+
+    private final int value;
+
+    NodeState(int value)
+    {
+        this.value = value;
+    }
+
+    public static NodeState valueOf(int value)
```

Comment:
Would annotation `ThriftEnum` and `ThriftEnumValue` achieve the same goal? See https://github.com/prestodb/drift/blob/d6a6da54864e90a09f2c8d12e798a3fc1a5df843/drift-codec/src/test/java/com/facebook/drift/codec/Fruit.java for an example :)

Reply:
This class is in SPI unfortunately.

Reply:
Update: I see the discussion in https://github.com/prestodb/presto/pull/13894/commits/8cb6c94418bf67308041a77cce45ac6332f460b2#r366518793 . Maybe just add comment explain why not using `ThriftEnum` :)